### PR TITLE
refactor: use kotlinx.serialization for json parsing

### PR DESCRIPTION
**Pull request by averen**
This pull request aims to replace the Klaxon JSON implementation with kotlinx.serialization.

### CODE REVIEW

File: src/main/kotlin/com/sandrabot/sandra/config/GuildConfig.kt [hunk 1]
```diff
@@ -16,16 +16,16 @@
...
-import com.sandrabot.sandra.entities.Locale
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
 
 /**
  * Stores Sandra-specific properties and settings for guilds.
  */
-class GuildConfig(val id: Long) : Configuration {
+@Serializable
+@SerialName("guild")
```
Comment (averen) [inline src/main/kotlin/com/sandrabot/sandra/config/GuildConfig.kt]: Serial name is no longer needed

File: src/main/kotlin/com/sandrabot/sandra/config/UserConfig.kt [hunk 2]
```diff
@@ -16,15 +16,16 @@
...
-import com.sandrabot.sandra.entities.Locale
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
 
 /**
  * Stores Sandra-specific properties and settings for users.
  */
-class UserConfig(val id: Long) : Configuration {
+@Serializable
+@SerialName("user")
```
Comment (averen) [inline src/main/kotlin/com/sandrabot/sandra/config/UserConfig.kt]: Serial name is no longer needed

File: src/main/kotlin/com/sandrabot/sandra/config/serializers.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+object ConfigSerializer : JsonContentPolymorphicSerializer<Configuration>(Configuration::class) {
+    override fun selectDeserializer(element: JsonElement) = when {
+        "prefixes" in element.jsonObject -> GuildConfig.serializer()
+        else -> UserConfig.serializer()
+    }
+}
+
+object ConfigTransformer : JsonTransformingSerializer<Configuration>(ConfigSerializer) {
+    override fun transformSerialize(element: JsonElement): JsonElement =
+        JsonObject(element.jsonObject.filterNot { (k, v) -> k == "type" })
```
Comment (averen) [inline src/main/kotlin/com/sandrabot/sandra/config/serializers.kt]: Closure parameter v is not used

File: src/main/kotlin/com/sandrabot/sandra/managers/ConfigurationManager.kt [hunk 4]
```diff
@@ -52,43 +48,36 @@ class ConfigurationManager(private val sandra: Sandra) : Service(5) {
...
-        val json = sandra.redis["$redisPrefix$id"]?.let {
-            klaxon.parseJsonObject(StringReader(it))
-        } ?: json { obj("id" to id) }
-        return (klaxon.fromJsonObject(json, type.java, type) as Configuration).also { cache[id] = it }
-    }
+    @OptIn(ExperimentalSerializationApi::class)
+    private fun getOrDefault(id: Long, redisPrefix: RedisPrefix) = (sandra.redis["$redisPrefix$id"]
+        ?.let { json.decodeFromString(ConfigSerializer, it) } ?: when (redisPrefix) {
+        RedisPrefix.GUILD -> GuildConfig(id)
+        RedisPrefix.USER -> UserConfig(id)
```
Comment (averen) [inline src/main/kotlin/com/sandrabot/sandra/managers/ConfigurationManager.kt:79]: Instances should not be explicitly instantiated

File: src/main/kotlin/com/sandrabot/sandra/utils/json.kt [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    is Iterable<*> -> JsonArray(this.map { it.toJsonElement() })
+    is Array<*> -> JsonArray(this.map { it.toJsonElement() })
+    else -> JsonPrimitive(this.toString()) // Or throw some "unsupported" exception?
+}
+
+fun JsonObject.obj(name: String): JsonObject? = get(name)?.escapeNull()?.jsonObject
+fun JsonObject.array(name: String): JsonArray? = get(name)?.escapeNull()?.jsonArray
+fun JsonObject.string(name: String): String? = get(name)?.escapeNull()?.jsonPrimitive?.contentOrNull
+
+fun JsonElement.escapeNull(): JsonElement? = if (this is JsonNull) null else this
```
Comment (averen) [inline src/main/kotlin/com/sandrabot/sandra/utils/json.kt]: Rename to escapeJsonNull

File: src/main/kotlin/com/sandrabot/sandra/managers/ConfigurationManager.kt (hunk 4 above)
Comment (averen) [reply src/main/kotlin/com/sandrabot/sandra/managers/ConfigurationManager.kt:79]: This can't really be helped unless we refactored again

**Review by Legolog3 — APPROVED** [review]
wow ifone (looks good to me)

**Pull request merged**